### PR TITLE
Added Prettier

### PR DESCRIPTION
**Pull request by jason-fox**
Update JavaScript code to follow prettier config.

* Add Prettier to package.json
* Run Linter
* Fixed outstanding issues

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
 
-var endpoint = process.env.TEST_ENDPOINT;
-var ngsild = 'ngsi-ld/v1';
+
+const endpoint = process.env.TEST_ENDPOINT;
+let ngsild = 'ngsi-ld/v1';
 
 // Just to test an old NGSIv2 endpoint
-if (process.env.FAKE_LD == 'yes') {
+if (process.env.FAKE_LD === 'yes') {
```
Comment (jmcanterafonseca) [inline common.js:16]: why are you removing the triple equals?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+[*]
+indent_style = tab
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{json,yml,md,babelrc,eslintrc,remarkrc}]
+indent_style = space
+indent_size = 2
```
Comment (jmcanterafonseca) [inline .eslintignore]: nit: new line needed

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "tamia",
+  "parserOptions": {
+    "sourceType": "module"
+  },
+  "rules": {
+    "valid-jsdoc": 0
+  }
+}
```
Comment (jmcanterafonseca) [inline .eslintrc]: nit: new line needed

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,41 @@
...
+*.pid
+*.seed
+*.pid.lock
+node_modules/
+jspm_packages/
+.npm
+.eslintcache
+*.tgz
+.env
+.next
```
Comment (jmcanterafonseca) [inline .gitignore:42]: too big list, is that really necessary?

File: .npmignore [hunk 5]
```diff
@@ -0,0 +1 @@
+__tests__
```
Comment (jmcanterafonseca) [inline .npmignore]: nit: new line needed

File: contextConsumption/query_entities_test.js [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+const testedResource = require('../common.js').testedResource;
+const http = require('../http.js');
+
+const entitiesResource = testedResource + '/entities/';
+const assertRetrievedQuery = require('../common.js').assertRetrievedQuery;
+const serializeParams = require('../common.js').serializeParams;
 
 describe('Query Entity. JSON. Default @context', () => {
-  let entity = {
+  const entity = {
```
Comment (jmcanterafonseca) [inline contextConsumption/query_entities_test.js:11]: is it really needed to change from let to const? 

From my point of view that is not a const just a variable in your code

File: package.json [hunk 7]
```diff
@@ -8,12 +8,23 @@
...
+    "ajv": "^6.5.2",
+    "eslint": "^5.6.0",
+    "eslint-config-tamia": "^6.0.1",
+    "eslint-plugin-prettier": "^2.6.2",
+    "jest": "^23.6.0",
+    "memfs": "2.9.4",
+    "prettier": "^1.14.2"
+  },
+  "engines": {
+    "node": ">=8.6"
```
Comment (jmcanterafonseca) [inline package.json:20]: are you really sure that this low version of Node is the minimal one? We are using features from Ecma2017.

**Review by jmcanterafonseca — CHANGES_REQUESTED** [review]

File: contextConsumption/query_entities_test.js (hunk 6 above)
Comment (jason-fox) [reply contextConsumption/query_entities_test.js:11]: NTC - its a variable that never changes - therefore its a `const`. 
Don't blame me - Its not me who amends the code - it is *prettier* changing it on the fly
I happen to agree with *Prettier*  by the way.

File: .gitignore (hunk 4 above)
Comment (jason-fox) [reply .gitignore:42]: This is the usual list I use for all common temporary file types.

`.eslintcache` is the essential addition of course

File: common.js (hunk 1 above)
Comment (jason-fox) [reply common.js:16]: NTC. I'm not removing the triple equals, I'm adding it. The old code had double equals which fails the linter.

Comment (jmcanterafonseca) [reply common.js:16]: oh, yes I didn't realise

File: package.json (hunk 7 above)
Comment (jason-fox) [reply package.json:20]: NTC. According to https://node.green/ ECMA2017 is 99% supported - are you planning on using shared memory atomics or `Array.prototype.keys`? Please don't. 😄 

I tend to use `8.6` as a low default for Node Carbon  8.x (I know we could be upgrading to  Node Dubnium 10.x LTS next month, but that would be another PR)

Would you prefer `8.12` LTS?

File: .eslintignore (hunk 2 above)
Comment (jason-fox) [reply .eslintignore]: Fixed a02a6607

File: .eslintrc (hunk 3 above)
Comment (jason-fox) [reply .eslintrc]: Fixed a02a6607

File: .npmignore (hunk 5 above)
Comment (jason-fox) [reply .npmignore]: Fixed a02a6607

**Comment by jmcanterafonseca** [comment]
@jason-fox 

I've applied the patch, then run `npm test` and tests run but at the end I'm getting the following error

```
Ran all test suites.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ngsi-ld_tests@1.0.0 test:jest: `jest`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ngsi-ld_tests@1.0.0 test:jest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jcantera/.npm/_logs/2018-09-27T12_34_37_042Z-debug.log
npm ERR! Test failed.  See above for more details.
```

**Comment by jason-fox** [comment]
> ```
> Ran all test suites.
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! ngsi-ld_tests@1.0.0 test:jest: `jest`
> npm ERR! Exit status 1
> ...etc
> ```

Fixed 57fd255 -  This is not an issue if running tests using `npm test` rather than `npm run ...` - see the intermediate travis build with 62 failing tests and a clean exit.

I've returned the travis build back to just running the linter for now (see 8b2c628)

File: package.json [hunk 8]
```diff
@@ -8,12 +8,22 @@
...
   },
   "scripts": {
+    "pretest": "npm run lint",
```
Comment (jmcanterafonseca) [inline package.json]: why do we need to run lint before testing? 

I don't agree.

**Review by jmcanterafonseca — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (jason-fox) [reply package.json]: Fixed 4838ecc - I usually add linting to `pretest` for all of the reasons described [here](https://softwareengineering.stackexchange.com/questions/308515/whats-the-point-of-running-unit-tests-on-a-ci-server)

* The developer doesn't have the discipline to lint manually prior to push
* They have forgotten to lint or test locally on their PR

I notice that travis is **already** running the linter as a `script` so we're already protected from laziness I see, so you are right, we don't need to lint twice. This repo is a bit of an edge case though, as the linter is only really there for formatting the test harness, not the source code like a 'real' project.

This does mean however that when running locally devs should remember to the following:

```console
npm run lint && npm test
```
to ensure they are running the tests against well formatted code

**Comment by jmcanterafonseca** [comment]
thanks @jason-fox! excellent work. now landing

**Review by jmcanterafonseca — APPROVED** [review]
